pageserver/src/tenant/size.rs

```diff
@@ -306,7 +322,33 @@ pub(super) async fn gather_inputs(
                 timeline_id: timeline.timeline_id,
                 kind,
             });
-            parent += 1;
+
+            parent = segments.len() - 1;
+
+            if kind == LsnKind::LeasePoint {
+                let mut lease_parent = parent;
+                segments.push(SegmentMeta {
+                    segment: Segment {
+                        parent: Some(lease_parent),
+                        lsn: lsn.0,
+                        size: None,
+                        needed: next_gc_cutoff <= lsn,
```

Comment:
This might need to be documented as well,

Reply:
addressed